app/services/search/postgres/podcast_episode.rb

```diff
@@ -0,0 +1,55 @@
+module Search
+  module Postgres
+    class PodcastEpisode
+      ATTRIBUTES = [
+        "podcast.id",
+        "podcast.image",
+        "podcast.published",
+        "podcast.slug",
+        "podcast_episodes.body",
+        "podcast_episodes.comments_count",
+        "podcast_episodes.id",
+        "podcast_episodes.podcast_id",
+        "podcast_episodes.processed_html",
+        "podcast_episodes.published_at",
+        "podcast_episodes.quote",
+        "podcast_episodes.reactions_count",
+        "podcast_episodes.subtitle",
+        "podcast_episodes.summary",
+        "podcast_episodes.title",
+        "podcast_episodes.website_url",
+      ].freeze
+      private_constant :ATTRIBUTES
+
+      DEFAULT_PER_PAGE = 60
+      private_constant :DEFAULT_PER_PAGE
+
+      MAX_PER_PAGE = 120 # to avoid querying too many items, we set a maximum amount for a page
+      private_constant :MAX_PER_PAGE
+
+      def self.search_documents(page: 0, per_page: DEFAULT_PER_PAGE, sort_by: nil, sort_direction: nil, term: nil)
+        # NOTE: [@rhymes/atsmith813] we should eventually update the frontend
+        # to start from page 1
+        page = page.to_i + 1
+        per_page = [(per_page || DEFAULT_PER_PAGE).to_i, MAX_PER_PAGE].min
+
+        relation = ::PodcastEpisode.includes(:podcast).where(podcast: { published: true })
+        relation = relation.search_podcast_episodes(term) if term.present?
```

Comment:
I see you've added the indexes, good call, as the query takes 5.5 seconds in production: https://dev.to/admin/blazer/queries/451-podcast-search

It takes ~ 80ms if we remove the ranking https://dev.to/admin/blazer/queries/452-podcast-search-no-ranking which goes back to 5 seconds if we move the ranking in the outer SELECT https://dev.to/admin/blazer/queries/453-podcast-search-ranking-outer-select

Adding a couple of more options in case the indexes don't help:

1. write the query manually as in [User.search_by_name_and_username](https://github.com/forem/forem/blob/0ec03063fdb4c5bb0d2df848f67bc9cd2adc6c24/app/models/user.rb#L269)
2. do step 1 adding ranking as `SELECT` column and order by that by default

:D

Reply:
Great idea! If it's too slow we'll probably explore both of these options 💯